android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/CarAppSyncComponent.kt

```diff
@@ -0,0 +1,164 @@
+package com.mapbox.navigation.examples.androidauto
+
+import androidx.car.app.Session
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.androidauto.ActiveGuidanceState
+import com.mapbox.androidauto.ArrivalState
+import com.mapbox.androidauto.FreeDriveState
+import com.mapbox.androidauto.MapboxCarApp
+import com.mapbox.androidauto.RoutePreviewState
+import com.mapbox.maps.logI
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.dropin.NavigationView
+import com.mapbox.navigation.dropin.NavigationViewListener
+import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+
+/**
+ * This is a temporarily solution for syncing two new libraries, Drop-in-ui and the Mapbox AA.
+ *
+ * The libraries are defining public apis so that there can be options to determine the experience
+ * while both the car and phone are displayed.
+ */
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class CarAppSyncComponent private constructor() : MapboxNavigationObserver {
+
+    private var navigationView: NavigationView? = null
+    private var session: Session? = null
+
+    fun setNavigationView(navigationView: NavigationView) {
+        this.navigationView = navigationView
```

Comment:
Why do you set it both here and in `onCreate`? Seems asymmetric. Because the first `onCreate` won't be invoked?

Reply:
`CarAppSyncComponent` will be a memory leak if `navigationView` is not set to `null`